terraform/Dockerfile

```diff
@@ -15,6 +15,17 @@ RUN curl -sL  https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terr
 
 # install terraform plugins
 WORKDIR /terraform
+
+ENV TERRAFORM_PROVIDER_AWSTAG_VERSION 0.1
+ENV TERRAFORM_PROVIDER_AWSTAG_HASH f2f1a153637f837ab521a4f1d0e236e08927354610495a199e4f67985de28a83
+RUN curl -sL https://github.com/dippynark/terraform-provider-awstag/releases/download/release-${TERRAFORM_PROVIDER_AWSTAG_VERSION}/terraform-provider-awstag_linux_amd64.zip > /tmp/terraform-provider-awstag.zip && \
+    sha256sum /tmp/terraform-provider-awstag.zip && \
+    echo "${TERRAFORM_PROVIDER_AWSTAG_HASH}  /tmp/terraform-provider-awstag.zip" | sha256sum -c && \
+    unzip /tmp/terraform-provider-awstag.zip && \
+    rm /tmp/terraform-provider-awstag.zip && \
+    mv terraform-provider-awstag /terraform/terraform-provider-awstag && \
+    chmod +x /terraform/terraform-provider-awstag
```

Comment:
Before we can merge this we need to verify the binary using GPG. I think it would make sense to use airworthy and verify both our custom awstag provider and the terraform binary. 

https://github.com/jetstack/airworthy